src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -2,19 +2,128 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
+using System.Collections.Immutable;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Test.Utilities;
 using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
 using Xunit;
+using Xunit.Abstractions;
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
+    [CollectionDefinition(Name)]
+    public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoadTestFixture>
+    {
+        public const string Name = nameof(AssemblyLoadTestFixtureCollection);
+        private AssemblyLoadTestFixtureCollection() { }
+    }
+
+    [Collection(AssemblyLoadTestFixtureCollection.Name)]
     public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
     {
+        private static readonly CSharpCompilationOptions s_dllWithMaxWarningLevel = new(OutputKind.DynamicallyLinkedLibrary, warningLevel: CodeAnalysis.Diagnostic.MaxWarningLevel);
+        private readonly ITestOutputHelper _output;
+        private readonly AssemblyLoadTestFixture _testFixture;
+        public DefaultAnalyzerAssemblyLoaderTests(ITestOutputHelper output, AssemblyLoadTestFixture testFixture)
+        {
+            _output = output;
+            _testFixture = testFixture;
+        }
+
+        [Fact, WorkItem(32226, "https://github.com/dotnet/roslyn/issues/32226")]
+        public void LoadWithDependency()
```

Comment:
It feels like the resources created by this test could be pushed into the fixture, especially because it's the same resources being created and used in two very similar tests (the other is in the ShadowLoaderTests)

Reply:
Done